### ISSUE TITLE
IE 11 polyfill broken with Nuxt 2.0 + apollo-module

### ISSUE THREAD

**Issue by dweldon** [nuxt-community/apollo-module#162]
### Version

[v4.0.0-rc.2](https://github.com/apollo-module/releases/tag/v4.0.0-rc.2)

### Reproduction link

[https://github.com/dweldon/nuxt-polyfill-issue](https://github.com/dweldon/nuxt-polyfill-issue)

### Steps to reproduce

git clone https://github.com/dweldon/nuxt-polyfill-issue.git
cd nuxt-polyfill-issue && npm i
npm start

Navigate to http://localhost:3000/

### What is expected ?

The page should render "Hello World"

### What is actually happening?

On IE 11, an error page is rendered. Object.entries is undefined.

### Additional comments?

I don't know if this is a bug with apollo-module, or with Nuxt 2.0, but the combination breaks on IE 11, and probably many other older browsers. If I remove the apollo-module code, Object.entries is defined. This implies one of: (1) Nuxt is executing core-js too late, (2) apollo-module or one of its dependencies is overwriting the polyfill.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/apollo-module/issues/c143">#c143</a>)</em></sub></div>

**Comment by clarkdo** [comment]
Sorry about that I have some personal stuff recently, I’ll look into it immediately when available.

**Comment by dohomi** [comment]
can you please check if `4.0.0.rc2.2` is fixing it for you? Are you using Nuxt v1 or v2?

**Comment by dohomi** [comment]
@dweldon I haven't experience it so far. But here is a polyfill stated: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#Polyfill
Do you know where that code is used? Because this package does not use Object.entries, something seems to be not right with the setup of Babel

**Comment by dweldon** [comment]
So I think progress it being made. If you do a clean checkout of the reproduction project (or remove `node_modules` and `package-lock.json`), when I build with `4.0.0-rc2.3`, I now see a spinner instead of an error page. The current error in IE11 is:

```
Expected identifier
vendors.app.js (5179,7)
```

In my build that line is:
`const { ApolloLink, Observable } = __webpack_require__(/*! apollo-link */ "./node_modules/apollo-link/lib/index.js")`

So Babel just isn't processing that file?

**Comment by Ayiga** [comment]
Looking into this issue, it stems directly from using [vue-cli-plugin-apollo](https://github.com/Akryum/vue-cli-plugin-apollo) specifically with it's dependency on [apollo-upload-client](https://github.com/jaydenseric/apollo-upload-client) located [here](https://github.com/Akryum/vue-cli-plugin-apollo/blob/b99182283eeb8ea49336bb1617a309269d32dc8b/graphql-client/src/index.js)

At least according to this output file that's what is being output with the `const` keywords:
![image](https://user-images.githubusercontent.com/3937524/47051962-bac15500-d163-11e8-8bd2-75d77279cc86.png)

It's a bit hard to follow because `vue-cli-plugin-apollo` is really mean to convert existing `Vue-Apollo` projects to allow for them to enable `SSR`.  However it's being included in this project as a convenience to prevent the user from having to mess around with the various `Link`s that `apollo` requires.  This ends up being a bit of a double-edged sword, because the result ends up making the packaged a lot bigger, and the cli is really meant to be used with `node` directly, but at the same time it's much easier to configure.

I'm going to keep looking into this, because I need an answer for it today.

**Comment by Leruiz** [comment]
I have met the same problem and solved it. 

Actually, the latest ```vue-cli-plugin-apollo``` depends on ```apollo-upload-client@9.0.0```. But the exported function ```createUploadLink ```   is written with ```es6``` in   ```src/index.js``` of ```apollo-upload-client@9.0.0```, which is not supported in ```ie 11```.  So the solution is very easy, to use a version of ```apollo-upload-client``` with codes written by ```es5```. I use ```apollo-upload-client@8.1.0```. To set the version, add resolutions into ```package.json``` as below:

`
"resolutions": {
    "apollo-upload-client": "8.1.0"
  },
`

Then the latest ```@nuxtjs/apollo``` can be used normally in ie11.

**Comment by dohomi** [comment]
Shouldn't babel of the configuration take care of transpiling? For me this looks like there is some configuration missing to enable IE11. But that would be rather a Nuxt issue itself than this plugin

**Comment by Leruiz** [comment]
In the webpack configurations of nuxt 2, files in ```node_modules``` are excluded by ```babel-loader``` at default. So the code of ```apollo-upload-client``` are not be transpiled. So, you have to add customized babel-loader configurations in ```nuxt.config.js``` if you want babel to handle it.

**Comment by Leruiz** [comment]
Now the author of ```apollo-upload-client``` has fixed this problem and released the latest version ```v9.1.0```.  This error disappears in my project.

**Comment by dweldon** [comment]
@Leruiz Thanks for the update.

When I upgrade all packages and add a polyfill for `Object.entries`, I can finally get it to run so I'll close this now.

**Comment by dweldon** [comment]
@janzheng Please see the commit history [here](https://github.com/dweldon/nuxt-polyfill-issue). I've kept the example up to date.

**Comment by azltdanh** [comment]
For google user who comes here, I am using `nuxt 2.x`, and the fix is quite simple, you just need to add transpile in `nuxt.config.js`

```
build: { transpile: ['vue-cli-plugin-apollo'] }
```
